### PR TITLE
Add viewroll and viewmodel bob tilt

### PR DESCRIPTION
**Pull request by sabian-heatwave**
Viewroll and viewmodel bob tilt was loved by many players at Half-Life's release but was removed from the engine in version 1.1.1.0. I think that adding this would attract more players to AG as it brings a removed old-school feature back.

Video demonstration:
https://www.youtube.com/watch?v=Yf2YzRkdVOU

### CODE REVIEW

File: cl_dll/view.cpp [hunk 1]
```diff
@@ -661,6 +664,7 @@ void V_CalcNormalRefdef ( struct ref_params_s *pparams )
 	view->angles[YAW]   -= bob * 0.5;
 	view->angles[ROLL]  -= bob * 1;
 	view->angles[PITCH] -= bob * 0.3;
+	VectorCopy(view->angles, view->curstate.angles);
```
Comment (YaLTeR) [inline cl_dll/view.cpp]: What is this for? Seems unrelated to the change.

Comment (sabian-heatwave) [reply cl_dll/view.cpp]: It makes the viewmodel bob go corner-to-corner as shown in the video instead of just back-and-forth, I think that's why it's commented "throw in a little tilt"

**Comment by sabian-heatwave** [comment]
> Sure, but it should be disabled by default.

@YaLTeR I will make that change now.

File: cl_dll/view.cpp (hunk 1 above)
Comment (YaLTeR) [reply cl_dll/view.cpp]: I'm asking specifically about your VectorCopy addition.

Comment (sabian-heatwave) [reply cl_dll/view.cpp]: The VectorCopy actually makes the bob go corner-to-corner (I think.)

Comment (YaLTeR) [reply cl_dll/view.cpp]: Could you remove it and check? I'd rather not introduce extra lines of code with little understanding of what they do.

File: cl_dll/view.cpp [hunk 2]
```diff
@@ -409,6 +411,7 @@ void V_CalcViewRoll ( struct ref_params_s *pparams )
 	viewentity = gEngfuncs.GetEntityByIndex( pparams->viewentity );
 	if ( !viewentity )
 		return;
+	pparams->viewangles[ROLL] = V_CalcRoll (pparams->viewangles, pparams->simvel, cl_viewrollangle->value, cl_viewrollspeed->value ) * 4;
```
Comment (YaLTeR) [inline cl_dll/view.cpp]: Should this be a += so as to not override the roll angle?

File: cl_dll/view.cpp (hunk 1 above)
Comment (sabian-heatwave) [reply cl_dll/view.cpp]: I already did that with [another SDK](https://github.com/thephoenixprojctsoftware/zamnhl). Removing it does not allow the bob to go corner-to-corner, the bob goes back-and-forth.

File: cl_dll/view.cpp (hunk 2 above)
Comment (sabian-heatwave) [reply cl_dll/view.cpp]: I've actually never tried that before.. will do now.

Comment (sabian-heatwave) [reply cl_dll/view.cpp]: Adding `+=` made no in-game difference

Comment (YaLTeR) [reply cl_dll/view.cpp]: Probably because there was no viewangle roll. I suggest then changing it to += because the code right below also does += and not =.

File: cl_dll/view.cpp (hunk 1 above)
Comment (YaLTeR) [reply cl_dll/view.cpp]: I checked myself and it does indeed make that difference. Alright.

**Comment by YaLTeR** [comment]
Thanks!
